scm/driver/gitea/git.go

```diff
@@ -24,6 +25,7 @@ func (s *gitService) FindBranch(ctx context.Context, repo, name string) (*scm.Re
 }
 
 func (s *gitService) FindCommit(ctx context.Context, repo, ref string) (*scm.Commit, *scm.Response, error) {
+	ref = strings.Replace(ref, "/", "%2F", -1)
```

Comment:
imho it would be safer to use `url.QueryEscape`
```suggestion
	ref = url.QueryEscape(ref)
```

Reply:
I think `PathEscape` should be used instead

Reply:
true :+1: